spec/controllers/books_controller_spec.rb

```diff
@@ -202,37 +186,17 @@
     context "when user is signed in and is an admin" do
       let(:user) { create(:user, admin: true) }
 
-      before :each do
+      it "deletes the book and returns 204 status" do
         sign_in user
-      end
-
-      context "when is succesfully destroyed" do
-        before :each do
-          @book1 = create(:book)
-          @book2 = create(:book)
-        end
-
-        it "deletes the book and returns 204 status" do
-          expect{ 
-            delete :destroy, params: { id: @book1.id }
-          }.to change(Book, :count).by(-1)
 
-          expect(response.status).to eq 204
-        end
-      end
+        @book1 = create(:book)
+        @book2 = create(:book)
 
-      context "when book doesn't exist" do
-        before :each do
-          delete :destroy, params: { id: 'not valid' }
-        end
+        expect{ 
+          delete :destroy, params: { id: @book1.id }
```

Comment:
we use the structure of putting the api called inside a before :each, and then each expect inside an it clause, it's cleaner that way